public/app/features/datasources/state/actions.ts

```diff
@@ -121,19 +152,9 @@ export const testDataSource = (
           path: editLink,
         });
       } catch (err) {
-        let message: string | undefined;
-        let details: HealthCheckResultDetails;
-
-        if (err instanceof HealthCheckError) {
-          message = err.message;
-          details = err.details;
-        } else if (isFetchError(err)) {
-          message = err.data.message ?? `HTTP error ${err.statusText}`;
-        } else if (err instanceof Error) {
-          message = err.message;
-        }
-
-        dispatch(testDataSourceFailed({ message, details }));
+        const formattedError = parseHealthCheckError(err);
+
+        dispatch(testDataSourceFailed({ ...formattedError }));
```

Comment:
What's the reason to pass in a copy of `formattedError` here?

Reply:
Sorry changed in the call above, but missed here. Thanks for catching it